gemm/src/kernels.cu

```diff
@@ -41,12 +77,37 @@ void sgemm(const int M,
            const float* const B,
            float* const C,
            const int V,
-           const int W) {
+           const int W,
+           int tile_dim) {
+    // The maximum memory for the RTX 4070 (Compute Capability 8.9) is 99KB
+    const size_t smem_size_max = 99 * 1024;
```

Comment:
Small nit: I think it's better to declare these "magic numbers" at the top of the file as global variables; e.g., 

```constexpr size_t SMEM_SIZE_MAX = 99*1024```